### ISSUE TITLE
IOS Project Generation Ignoring Name and Version

### ISSUE THREAD

**Issue by Braden1996** [expo/expo#17324]
### Summary

I am upgrading my project from SDK 44 to SDK 45 (beta 9). However, when I execute `expo run:ios` the generated XCode project doesn't have the correct name and version.
![image](https://user-images.githubusercontent.com/5165963/166554541-b7de0af8-e921-48e6-879e-53be8c672cca.png)
![image](https://user-images.githubusercontent.com/5165963/166554557-45b326af-1384-425c-9ff6-43c59e0f4c23.png)

### SDK Version

45

### Environment

      Android Studio: 2020.3 AI-203.7717.56.2031.7935034
      Xcode: 13.3.1/13E500a - /usr/bin/xcodebuild
    npmPackages:
      react-native: 0.68.1 => 0.68.1
    npmGlobalPackages:
      eas-cli: 0.52.0
      expo-cli: 5.4.3
    Expo Workflow: bare

### Reproducible demo

I tried to create a reproducible repo with a similar layout (see https://github.com/Braden1996/expo-45-issue). However, it doesn't showcase the issue. I'm not sure how it's meaningfully different to my local setup.

**Comment by brentvatne** [comment]
@Braden1996 - hi there! if you're not able to reproduce in a new repository then you likely have some config plugin that is changing the behavior. try using this approach to narrow down the source of the problem: https://expo.fyi/manual-debugging

**Comment by Braden1996** [comment]
> @Braden1996 - hi there! if you're not able to reproduce in a new repository then you likely have some config plugin that is changing the behavior. try using this approach to narrow down the source of the problem: https://expo.fyi/manual-debugging

Hey @brentvatne, thanks for getting back so fast. I figured out what the issue was and why I wasn't able to easily reproduce.

The issue was our `yarn.lock` was pinning `@expo/config-plugins` to version `^4.0.15`. Quite a few Expo packages (e.g. `expo-community-flipper`) haven't been updated to point to the newer `4.1.4` - so this regression wasn't resolved by Yarn's hoisting. The impact of this was that our prebuild process was throwing because the IOS Paths code wasn't looking for `AppDelegate.mm`. On consequent runs, however, an AppDelegate could be found - but the initial project generation had been corrupted by this point.

A full wipe `yarn.lock` isn't feasible on my particular project, so I manually modified the lock by removing the old version entries and adding this:
![CleanShot 2022-05-04 at 12 24 59](https://user-images.githubusercontent.com/5165963/166672542-c07d705d-e659-422d-9b56-08896e5111be.png)
